searx/webapp.py

```diff
@@ -490,6 +491,27 @@ def index():
     Supported outputs: html, json, csv, rss.
     """
 
+    def hash_client(request):
+        _ = hashlib.sha256()
+        _.update(request.headers.get('User-Agent').encode('utf-8'))
```

Comment:
Using the User-Agent for the hash is a very bad idea because in the Searx community using a user agent switcher extension is quite common so this will most likely annoy these people if they get redirected too often.

Reply:
good point / thanks!

I really like to tag the application behind a query, not only the IP where it comes from. Did you have any idea which data might be useful to identify the application?